org-code-javabuilder/lib/src/main/java/org/code/neighborhood/GridSquare.java

```diff
@@ -0,0 +1,119 @@
+package org.code.neighborhood;
+import org.json.simple.JSONObject;
+
+public class GridSquare {
+    private String color;
+    private boolean passable;
+    private int paintCount;
+    private enum SquareType{ WALL, OPEN, START, FINISH, OBSTACLE, STARTANDFINISH, UNKNOWN};
+    private SquareType squareType;
+
+    public GridSquare(JSONObject descriptor) {
```

Comment:
This class would be more decoupled and more unit testable if the constructor took primitive types instead of the JSONObject.